classes/notification/form/PKPNotificationsUnsubscribeForm.inc.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * @defgroup notification_form Notification Form
+ */
+
+/**
+ * @file classes/notification/form/PKPNotificationsUnsubscribeForm.inc.php
+ *
+ * Copyright (c) 2014-2019 Simon Fraser University
+ * Copyright (c) 2000-2019 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class PKPNotificationsUnsubscribeForm
+ * @ingroup notification_form
+ *
+ * @brief Form to unsubscribe from email notifications.
+ */
+
+
+import('lib.pkp.classes.form.Form');
+
+class PKPNotificationsUnsubscribeForm extends Form {
+	var $_notification;
+	var $_validationToken;
+
+	/**
+	 * Constructor.
+	 */
+	function __construct($notification, $validationToken) {
+		parent::__construct('notification/unsubscribeNotificationsForm.tpl');
+
+		// Validation checks for this form
+		$this->addCheck(new FormValidatorPost($this));
+		$this->addCheck(new FormValidatorCSRF($this));
+
+		$this->_notification = $notification;
+		$this->_validationToken = $validationToken;
+	}
+
+	/**
+	 * Assign form data to user-submitted data.
+	 */
+	function readInputData() {
```

Comment:
For new code, define visibility (`public`)

Reply:
Fixed